src/sentry/notifications/manager.py

```diff
@@ -389,3 +390,34 @@ def disable_settings_for_users(
                     target_id=user.actor_id,
                     defaults={"value": NotificationSettingOptionValues.NEVER.value},
                 )
+
+    def has_any_provider_settings(
+        self, recipient: Team | User, provider: ExternalProviders
+    ) -> bool:
+        # Explicitly typing to satisfy mypy.
+        has_settings: bool = (
+            self._filter(provider=provider, target_ids={recipient.actor_id})
```

Comment:
@mgaeta How does this logic know to check a user vs a team? Couldn't the IDs collide?

Reply:
The actor ID space is shared by users and teams.